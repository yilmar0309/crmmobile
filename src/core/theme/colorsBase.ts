export const colorsBase = {
  WHITE: '#FFFFFF',
  BLACK: '#000000',
  GRAY: '#F8F8F8',
};
