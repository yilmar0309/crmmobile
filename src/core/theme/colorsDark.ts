export const colorsDark = {
  BACKGROUND: '#2C3E50',
  ON_BACKGROUND: '#E1E2E1',

  // Background primary color
  PRIMARY_COLOR: '#4d9ecf',
  ON_PRIMARY: '#FFFFFF',
  SECONDARY_COLOR: '#356ab0',

  // Background
  BACKGROUND_SCREEN_COLOR: '#2C3E50',

  // SURFACE
  SURFACE_1: '#34495E',

  // CONTENT
  CONTENT_PRIMARY: '#E1E2E1',
  CONTENT_SECONDARY: '#AAB0B6',

  // Text color
  PRIMARY_TEXT_COLOR: '#E1E2E1',
  SECONDARY_TEXT_COLOR: '#AAB0B6',
  DISABLED_TEXT_COLOR: '#52555A',
  TEXT_SIMPLE_COLOR: '#34495E',

  // TextInput color
  BACKGROUND_TEXT_INPUT_COLOR: '#34495E',

  // Button
  BACKGROUND_BUTTON_COLOR: '#2C3E50',
  BUTTON_TEXT_CONTAINED_COLOR: '#FFFFFF',
  BUTTON_TEXT_OUTLINED_COLOR: '#67b7ed',
  BUTTON_TEXT_COLOR: '#FFFFFF',
  BUTTON_BORDER_COLOR: '#67b7ed',
  BUTTON_DISABLED_COLOR: '#AAB0B6',
  BUTTON_DISABLED_TEXT_COLOR: '#CDD1D0',

  // Bottom Tab
  BOTTOM_TAB_COLOR: '#E1E2E1',

  // GRAY
  GRAY_01: '#34495E',
  GRAY_02: '#273746',
  GRAY_03: '#1B2C3B',
  GRAY_04: '#22313F',
  GRAY_05: '#293B4D',
  GRAY_06: '#999999',

  // Alert color
  SUCCESS_COLOR: '#2D7D46',
  WARNING_COLOR: '#C97B00',
  ERROR_COLOR: '#D0342C',
  INFO_COLOR: '#8D8B92',

  WHATSAPP: '#1E7D47',
  FACEBOOK: '#1D508D',
  INSTAGRAM: '#AD2671',
  TIKTOK: '#181818',

  // Basic color
  WHITE: '#FFFFFF',
  BLACK: '#121212',

  OUTLINE: '#2C313C',

  // SENTIMENT
  SENTIMENT_SUCCESS_BACKGROUND: '#1B4026',
  SENTIMENT_SUCCESS: '#2D7D46',
  SENTIMENT_INFO_BACKGROUND: '#2C3E50',
  SENTIMENT_INFO: '#1E2620',
  SENTIMENT_WARNING_BACKGROUND: '#332B1A',
  SENTIMENT_WARNING: '#C97B00',
  SENTIMENT_NEGATIVE_ERROR_BACKGROUND: '#401311',
  SENTIMENT_NEGATIVE_COLOR: '#D0342C',
  SENTIMENT_TEXT: '#1E2620',
  SENTIMENT_POSITIVE_BG: '#1B4026',
  SENTIMENT_WARNING_BACKGROUND_LOCK: '#664D1A',
  SENTIMENT_WARNING_ON_WARNING_BG: '#3A2800',

  // MAPS
  COLOR_MAP_LOSSES: '#FF6B6B',
  COLOR_MAP_FOUND: '#4ECDC4',
  COLOR_MAP_ADOPTION: '#FFD93D',
};
