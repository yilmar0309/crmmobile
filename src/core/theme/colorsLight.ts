export const colorsLight = {
  BACKGROUND: '#F4F6F6',
  ON_BACKGROUND: '#191C1B',

  // Background primary color
  PRIMARY_COLOR: '#67b7ed',
  ON_PRIMARY: '#FFFFFF',
  SECONDARY_COLOR: '#4981dd',

  // Background
  BACKGROUND_SCREEN_COLOR: '#FFFFFF',

  // SURFACE
  SURFACE_1: '#FFFFFF',

  // CONTENT
  CONTENT_PRIMARY: '#191C1B',
  CONTENT_SECONDARY: '#3F4947',

  // Text color
  PRIMARY_TEXT_COLOR: '#263238',
  SECONDARY_TEXT_COLOR: '#757575',
  DISABLED_TEXT_COLOR: '#BABBBF',
  TEXT_SIMPLE_COLOR: 'rgba(44, 62, 80, 0.2)',

  // TextInput color
  BACKGROUND_TEXT_INPUT_COLOR: '#F8F8F8',

  // Button
  BACKGROUND_BUTTON_COLOR: '#FFFFFF',
  BUTTON_TEXT_CONTAINED_COLOR: '#FFFFFF',
  BUTTON_TEXT_OUTLINED_COLOR: '#67b7ed',
  BUTTON_TEXT_COLOR: '#000000',
  BUTTON_BORDER_COLOR: '#67b7ed',
  BUTTON_DISABLED_COLOR: '#CDD1D0',
  BUTTON_DISABLED_TEXT_COLOR: '#AAB0B6',

  // Bottom Tab
  BOTTOM_TAB_COLOR: '#000000',

  // GRAY
  GRAY_01: '#F8F8F8',
  GRAY_02: '#CDD1D0',
  GRAY_03: '#777B85',
  GRAY_04: '#999999',
  GRAY_05: '#F0F0F0',
  GRAY_06: '#999999',

  // Alert color
  SUCCESS_COLOR: '#36B37E',
  WARNING_COLOR: '#EAB55C',
  ERROR_COLOR: '#DC6358',
  INFO_COLOR: '#A3A1AF',

  WHATSAPP: '#25D366',
  FACEBOOK: '#3B5998',
  INSTAGRAM: '#C13584',
  TIKTOK: '#010101',

  // Basic color
  WHITE: '#FFFFFF',
  BLACK: '#000000',

  OUTLINE: '#BEC9C6',

  // SENTIMENT
  SENTIMENT_SUCCESS_BACKGROUND: '#F5FFF3',
  SENTIMENT_SUCCESS: '#006D3A',
  SENTIMENT_INFO_BACKGROUND: '#FFFFFF',
  SENTIMENT_INFO: '#00210D',
  SENTIMENT_WARNING_BACKGROUND: '#FFF8EF',
  SENTIMENT_WARNING: '#EDC843',
  SENTIMENT_NEGATIVE_ERROR_BACKGROUND: '#FFF8F7',
  SENTIMENT_NEGATIVE_COLOR: '#BA1A1A',
  SENTIMENT_TEXT: '#00210D',
  SENTIMENT_POSITIVE_BG: '#7FFBA8',
  SENTIMENT_WARNING_BACKGROUND_LOCK: '#FFE07E',
  SENTIMENT_WARNING_ON_WARNING_BG: '#231B00',

  // MAPS
  COLOR_MAP_LOSSES: '#FF6B6B',
  COLOR_MAP_FOUND: '#4ECDC4',
  COLOR_MAP_ADOPTION: '#FFD93D',
};
