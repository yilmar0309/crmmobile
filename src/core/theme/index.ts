export {colorsBase} from './colorsBase';
export {colorsLight} from './colorsLight';
export {colorsDark} from './colorsDark';
