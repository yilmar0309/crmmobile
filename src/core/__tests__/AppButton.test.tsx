import React from 'react';
import {StyleSheet, Text} from 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import {
  AppButton,
  AppButtonSizeVariant,
  AppButtonVariant,
} from '../components/AppButton';
import {colorsLight} from '../theme';

const MockIcon = () => <Text>Icon</Text>;

describe('AppButton', () => {
  it('renders correctly with minimal props', () => {
    const {getByText, toJSON} = render(
      <AppButton
        variant={AppButtonVariant.contained}
        size={AppButtonSizeVariant.medium}
        label="Click me"
      />,
    );

    expect(getByText('Click me')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('handles onPress events', () => {
    const onPressMock = jest.fn();
    const {getByText} = render(
      <AppButton
        variant={AppButtonVariant.contained}
        size={AppButtonSizeVariant.medium}
        label="Click me"
        onPress={onPressMock}
      />,
    );

    fireEvent.press(getByText('Click me'));
    expect(onPressMock).toHaveBeenCalled();
  });

  it('shows loading indicator when isLoading is true', () => {
    const {getByTestId} = render(
      <AppButton
        variant={AppButtonVariant.contained}
        size={AppButtonSizeVariant.medium}
        isLoading={true}
      />,
    );
    expect(getByTestId('app-button')).toBeTruthy();
  });

  it('shows disabled state correctly', () => {
    const {getByText} = render(
      <AppButton
        variant={AppButtonVariant.contained}
        size={AppButtonSizeVariant.medium}
        label="Disabled"
        disabled={true}
      />,
    );

    const textComponent = getByText('Disabled');
    const style = StyleSheet.flatten(textComponent.props.style);

    expect(style.color).toBe(colorsLight.BUTTON_DISABLED_TEXT_COLOR);
  });

  it('renders icon when provided', () => {
    const {getByText} = render(
      <AppButton
        variant={AppButtonVariant.contained}
        size={AppButtonSizeVariant.medium}
        iconLeft={<MockIcon />}
        label="With Icon"
      />,
    );

    expect(getByText('Icon')).toBeTruthy();
  });

  it('applies styles based on variant and size', () => {
    const {getByText} = render(
      <AppButton
        variant={AppButtonVariant.outlined}
        size={AppButtonSizeVariant.large}
        label="Test"
      />,
    );

    const textComponent = getByText('Test');
    const style = StyleSheet.flatten(textComponent.props.style);
    expect(style.fontSize).toBe(16);
  });
});
