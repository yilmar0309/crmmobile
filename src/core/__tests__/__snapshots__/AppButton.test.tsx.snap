// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppButton renders correctly with minimal props 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#67b7ed",
      "borderColor": "transparent",
      "borderRadius": 24,
      "borderWidth": 0,
      "justifyContent": "center",
      "minHeight": 40,
      "opacity": 1,
      "transform": [
        {
          "scale": 1,
        },
      ],
      "width": "100%",
    }
  }
  testID="app-button"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderRadius": 24,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      center={true}
      height={24}
      style={
        [
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          {},
          {},
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      width={24}
    />
    <Text
      style={
        [
          {
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": 0.5,
          },
          {
            "color": "#FFFFFF",
          },
          {
            "marginHorizontal": 8,
          },
        ]
      }
    >
      Click me
    </Text>
    <View
      center={true}
      height={24}
      style={
        [
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          {},
          {},
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      width={24}
    />
  </View>
</View>
`;
